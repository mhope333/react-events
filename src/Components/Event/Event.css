.ui.button.purple {
  background-color: rgb(100, 52, 128);
  color: white;
}
.ui.modal>.header:not(.ui) {
  font-size: 2.5rem;
  background-color: rgb(100, 52, 128);
  color: white;
}
.modalContent {
  font-size: small;
}
.registerButton {
  padding-bottom: 30px;
}