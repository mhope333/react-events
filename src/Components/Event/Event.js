import React from 'react'
import trainingIcon from '../../images/trainingIcon.png';
import marketingIcon from '../../images/marketingIcon.png';
import seminarIcon from '../../images/seminarIcon.png';
import {Button, Modal, Image, Header, Icon, Card} from 'semantic-ui-react'
import './Event.css';

function dateTimeConverter(unixTimestamp) {
  const dateObj = new Date(unixTimestamp * 1000); 
  const utcString = dateObj.toUTCString(); 
  const utcDateString = utcString.substring(0, utcString.length-4); // removes GMT
  return utcDateString;
}

function iconSelector(category) {
  if (category === 'Seminars') {
    return seminarIcon;
  } else if (category === 'Trainings') {
    return trainingIcon;
  } else if (category === 'Marketing') {
    return marketingIcon;
  } else {
    return trainingIcon;
  }
}

const event = (props) => {

  const category = props.event.category["1"].title;

  const favouriteEvents = props.favourites;
  const matchEventInFavourites = favouriteEvents.filter(event => event.id === props.event.id);
  const isFavourite = matchEventInFavourites.length === 1 ? true : false;

  return (
    <Card>
      <Card.Content>
        <Image
          floated='right'
          size='mini'
          src={iconSelector(category)} alt="Icon" className="ui rounded image"
        />
        <Card.Header>{props.event.name.en || props.event.name.fi}</Card.Header>
        <Card.Meta>{category}</Card.Meta>
        <Card.Description>Start time: {dateTimeConverter(props.event.start_time)}</Card.Description>
      </Card.Content>
      <Card.Content extra>
        <Modal trigger={<Button floated='left' className="purple">More Info</Button>} closeIcon>
          <Modal.Header className="modalHeader">Event Information</Modal.Header>
          <Modal.Content image>
            <Image wrapped size='medium' src={iconSelector(category)}/>
            <Modal.Description>
              <Header as="h2">{props.event.name.en || props.event.name.fi}</Header>
              <div className="modalContent">
                <p>Location: {props.event.location.en || props.event.location.fi}</p>
                <p>Start time: {dateTimeConverter(props.event.start_time)}</p>
                <p>Timezone: {props.event.timezone}</p>
                <p>Enrollment deadline: {dateTimeConverter(props.event.enrollment_deadline_utc)}</p>
              </div>
            </Modal.Description>
          </Modal.Content>
          <Modal.Actions>
            <div className="registerButton">
              <a href={props.event.enrollment_url} target="_blank" rel="noopener noreferrer">
                <Button floated='right' className="purple">Register</Button>
              </a>
            </div>
          </Modal.Actions>
        </Modal>
        <Button floated='right' onClick={() => props.toggleFavourite(props.event.id)} basic icon>
          {isFavourite ? <Icon color='yellow' name="star"/> : <Icon color='yellow' name="star outline"/>}
        </Button>
      </Card.Content>
    </Card>
  )
}

export default event;
