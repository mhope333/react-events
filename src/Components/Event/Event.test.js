import React from 'react';
import Event from './Event';
import {cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

const event = {
  id: '123',
  category: {
    '1': {
      id: 13341,
      title: "title"
    }
  },
  name: {
    en: 'eventName'
  },
  start_time: 1595829600,
  enrollment_deadline_utc: 1595829600,
  location: {
    en: 'location'
  },
  timezone: 'timezone',
  enrollment_url: 'www.someUrl.com'
};
const favourites = [];
function mockFunction() {
  return 'mock';
}

afterEach(() => {
  cleanup();
});

describe('Event component', () => {
  it('html should match snapshot', () => {
    const tree = renderer.create(<Event key={event.id} event={event} tooggleFavourite={mockFunction} favourites={favourites}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // TO DO: 
  // - add test for dateTimeConverter function
  // - add test for iconSelector function
  // - add unit tests for Event-List (With data and error scenario)
});
