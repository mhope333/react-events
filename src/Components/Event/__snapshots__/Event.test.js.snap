// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event component html should match snapshot 1`] = `
<div
  className="ui card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <img
      alt="Icon"
      className="ui mini right floated image ui rounded image"
      src="trainingIcon.png"
    />
    <div
      className="header"
    >
      eventName
    </div>
    <div
      className="meta"
    >
      title
    </div>
    <div
      className="description"
    >
      Start time: 
      Mon, 27 Jul 2020 06:00:00
    </div>
  </div>
  <div
    className="extra content"
  >
    <button
      className="ui left floated button purple"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      More Info
    </button>
    <button
      className="ui basic icon right floated button"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="yellow star outline icon"
        onClick={[Function]}
      />
    </button>
  </div>
</div>
`;
